site/docs/intro/under-the-hood.md

```diff
@@ -2,5 +2,54 @@
 sidebar_position: 4
 title: Under the hood
 ---
+## Provisioning preview environments
 
-# Under the hood
+When provisioning a new environment using the `up` command, `preevy` does the following:
+- Reads for default configuration and relevant keys from the current profile store.
+- Calculates environment name based on git branch or uses the `--id` flag.
+- Uses the local aws configuration to provision a new Lightsail VM.
+- Reads SSH keypair from profile to access the VM, if necessary, generate a new one.
+- Connects to the VM using SSH and setup Docker.
+- Reads the compose file and copies local volume mounts to the VM.
+- Augments the compose deployment with a helper service, `tunnel-agent`, responsible for connecting to the [tunnel server](/tunnel-server/overview.md).
+- Runs the application using Docker Compose with `--build` while using the local build context.
+- The `tunnel-agent` is inspecting the network configuration of all deployed services and create a tunnel for each service.
+- Fetch the urls from tunnel-agent and output them to the end user.
+
+## Profile configuration
+
+Preevy profile provide a mechanism for storing and sharing configuration and keys between different machines (multiple developers and/or CI Jobs).
+
+The profile storage consists of multiple tar archives that can be stored locally or in a remote storage provider. (S3 is the only supported remote storage implementation at the moment)
+
+:::note
+Profile store doesn't contains any cloud provider credentials.  
+When using `preevy` CLI, the user need to have a local cloud configuration (using `aws login` or `aws configure`, or environment variables) with the required permissions.
```

Comment:
maybe add a link to the required permissions?

Reply:
will add